### PR TITLE
support multiple groupIds in put endpoint

### PR DESCRIPTION
**Pull request by bingzhang**
## Description
_Please provide a summary of the pull request and the issue it fixes. Please add necessary details, context, dependencies, explanation of when review is needed (see next section), etc._

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

File: eventservice/api/controllers/events.py [hunk 1]
```diff
@@ -650,8 +652,11 @@ def event_operation_permission_check(req_data=None, event_id=None):
             # If this is a group event, apply group authorization. Regular events can proceed like before.
             if not check_group_event_admin_access(event, group_memberships):
                 abort(401)
+            if req_data and req_data.get('groupIds') != event.get('groupIds'):
```
[the remark this answers is not in the document]
Comment (bingzhang) [reply eventservice/api/controllers/events.py]: fixed

[the remark this answers is not in the document]
Comment (sandeep-ps) [reply eventservice/api/controllers/events.py]: Thanks. Re-reviewing.

File: eventservice/api/controllers/events.py [hunk 2]
```diff
@@ -650,8 +652,11 @@ def event_operation_permission_check(req_data=None, event_id=None):
             # If this is a group event, apply group authorization. Regular events can proceed like before.
             if not check_group_event_admin_access(event, group_memberships):
                 abort(401)
+            if req_data and set(req_data.get('groupIds')) != set(event.get('groupIds')):
```
Comment (sandeep-ps) [inline eventservice/api/controllers/events.py:655]: This check fails for events that do not contain the `groupIds` field. Please fix this and let me know after you have tested different scenarios. Thanks.

Comment (bingzhang) [reply eventservice/api/controllers/events.py:655]: I have some questions about it. events without groupIds should does not have 'createdByGroupId', then it is handled by the code if not event.get('createdByGroupId') and not event.get('groupIds'):

I am not sure it is right or not.

Comment (sandeep-ps) [reply eventservice/api/controllers/events.py:655]: This is for backward compatibility, i.e., current events with `createdByGroupId` and no `groupIds` in the data. Those events will not have `groupIds` and will throw a runtime error as the code tries to access the `groupIds` field in the above line, which doesn't exist. As these are optional fields, an event can contain either of these (associated with a single group vs. multiple groups) or both (groupIds should get priority in that case, I think), or none (events from sources like WebTools, which do not have any group associated with it).

Comment (bingzhang) [reply eventservice/api/controllers/events.py:655]: I see. shall we use or instead of and in that if condition?

Comment (bingzhang) [reply eventservice/api/controllers/events.py]: I updated the code after me and Sandeep discussed on it.

File: eventservice/api/controllers/events.py
Comment (sandeep-ps) [inline eventservice/api/controllers/events.py:1]: Could you please add some comments for the newly added conditions to clarify which events they are targeting? The rest looks good to me. Thanks.

Comment (bingzhang) [reply eventservice/api/controllers/events.py:1]: update comments.

File: eventservice/api/controllers/events.py (hunk 2 above)
Comment (sandeep-ps) [reply eventservice/api/controllers/events.py]: Thanks.

**Pull request merged**